VERSION.txt

```diff
@@ -1 +1 @@
-"0.6.5"
+"0.7.0"
```

Comment:
This is not a major version bump, btw.

Reply:
I think for pre-1.0 releases, major/minor doesn't have the same meaning, but generally bumping the minor is treated similarly to bumping the major in a post-1.0 release?

You could argue that magicl should already have been on a post-1.0 release, but that's a different story.

In practice I don't know how many downstream consumers magicl has besides quilc/qvm, so maybe it doesn't matter much, but I would lean towards keeping it pre-1.0.0 for at least a few more releases to give time for the new API to settle and shake out bugs before going to 1.0.

It just seems weird for something so new and not quite "stable" yet to get tagged as the 1.0 release. That is my $0.02 anyway, but I also would not ragequit if we made this 1.0.

Reply:
Can't stay 0.x forever. Shaking out bugs in a big release is what we're supposed to be doing now? i.e. what if magicl was already 1.x and we wanted to make this release? You wouldn't then say we should make a minor 1.y release to shake out the bugs and only then do a 2.0? Or would you? Genuine question!

Reply:
I don't know if it's what we *supposed* to be doing, but the reality is:

1) It'd be surprising if such massive changes didn't have a few bugs that need shaking out. See @kilimanjaro's comments above from just a quick run-through of the high-level docs.

2) There are quite a few comments in this PR suggesting API additions and improvements for follow-on PRs, so it doesn't quite seem like the API is that stable yet, which what 1.0 implies to my mind.

I guess my view is that a 1.0 release should be approached asymptotically rather than in a seismic shift.

If magicl already had a 1.0 release, then this would be a 2.0-worthy change for sure, but it hasn't so it feels odd to me to label this the 1.0 release based on the fact that it introduces major breaking changes, rather than basing it on the fact that the API feels in some sense "stable enough to be called 1.0", although the quotes there mean I don't have some clear criteria for what "stable enough" would mean.

Reply:
I don't like that "1.0.0" is overloaded with symbolism. Does semantic versioning just not apply for 0.y.z?

Not that it matters. I don't mind if we go to 0.7.0 rather than 1.0.0.

Reply:
I am not a semver expert, I just play one on TV, but according to their faq pre-1.0 releases are treated differently

https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

Of course, the very next bullet point makes the argument that magicl should probably have already been on a post-1.0 release...

I think one thing we both agree on is that it probably doesn't matter much if quilc and qvm are the only downstream consumers :)

Reply:
I will be bumping to 1.0.0

Reply:
I'm more or less with appleby on this one as well. Version 1.0.0 feels special because it is special. My only objection to _this_ being 1.0.0 is that we don't have much working experience with the API, outside of this code review, and so it is hard to anticipate what follow-up work will be on the "wish list".